apps/src/lib/kits/maker/boards/circuitPlayground/CircuitPlaygroundBoard.js

```diff
@@ -92,33 +96,50 @@ export default class CircuitPlaygroundBoard extends EventEmitter {
    */
   connectToFirmware() {
     return new Promise((resolve, reject) => {
-      const name = this.port_ ? this.port_.comName : undefined;
-      const serialPort = CircuitPlaygroundBoard.openSerialPort(name);
-      const playground = CircuitPlaygroundBoard.makePlaygroundTransport(
-        serialPort
-      );
-      const board = new five.Board({io: playground, repl: false, debug: false});
-      board.once('ready', () => {
-        this.serialPort_ = serialPort;
-        this.logWithFirehose(
-          'serial-port-set',
-          JSON.stringify({serialPort, name})
+      if (isWebSerialPort(this.port_)) {
+        const name = this.port_.getInfo().usbProductId;
+        CircuitPlaygroundBoard.openSerialPortWebSerial(this.port_).then(
+          port => {
+            this.initializePlaygroundAndBoard(port, name, resolve, reject);
+          }
         );
+      } else {
```

Comment:
As this work continues, it'd be great to think about how we can build an architecture that avoids a lot of `if webserial then ... else ...` in the same places.  

It seems it would be good to head towards an architecture where the serial interface implementation exists independent of business logic above it and hardware specific logic below it.  

Perhaps we could get to the point where we have a common interface but it can be instantiated as webserial vs. not webserial, one implementation of the interface for each system, and the code above it and below it doesn't have to know which one we're using.  And then, eventually, we can even remove the non-webserial implementation entirely.

Does that make sense as a general direction?

Reply:
Yeah, for sure. I think we're heading in that direction now I'm working on a wrapper for the WebSerialPorts to "look" like the Node Serial Port. If that pans out moving forward, it will make sense to pull that work back towards this original connection work and see how many of these if/elses we can pull.